simd/unit_tests/include/TestSIMD_Condition.hpp

```diff
@@ -0,0 +1,85 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_TEST_SIMD_CONDITION_HPP
+#define KOKKOS_TEST_SIMD_CONDITION_HPP
+
+#include <Kokkos_SIMD.hpp>
+#include <SIMDTesting_Utilities.hpp>
+
+template <typename Abi, typename DataType>
+inline void host_check_condition() {
+  auto a = Kokkos::Experimental::condition(
+      Kokkos::Experimental::simd<std::int32_t, Abi>(1) > 0,
+      Kokkos::Experimental::simd<std::uint64_t, Abi>(16),
+      Kokkos::Experimental::simd<std::uint64_t, Abi>(20));
+  EXPECT_TRUE(all_of(a == decltype(a)(16)));
+}
+
+template <typename Abi, typename... DataTypes>
+inline void host_check_condition_all_types(
+    Kokkos::Experimental::Impl::data_types<DataTypes...>) {
+  (host_check_condition<Abi, DataTypes>(), ...);
```

Comment:
Just for a clarification, unit tests for `Condition`, `Conversions` and `MaskOps` have not been adjusted to test with all data types yet, so that's why `DataType` is there but not used anywhere. It's a remnant of splitting away from what was initially used for MathOps testing.

I'll remove these currently unused testing structure of testing with all data types for now and use tuples instead for `Condition` unit test in a follow up refactoring.

Reply:
yeah

Reply:
Removed unnecessary structure in `Condition`, `Conversions`, and `MaskOps` for testing with all data types for now. [d540636](https://github.com/kokkos/kokkos/pull/6278/commits/d540636225baf102f84a2070b09666a1fd755d11)